src/js/Maze.js

```diff
@@ -51,6 +51,42 @@ class Maze {
             return index ===cells.length ? false : true;
         }
     }
+
+    /**
+     * Calculate distance of all maze cells from cell at given index,
+     * then use color to visualize distance.
+     *
+     * @param {number} cellIdx - Index of selected start cell.
+     */
+    colorByDistanceFromCell(cellIdx) {
+        const cells = this.getCells();
+        cells.forEach((cell) => cell.distance = null);
+
+        const startCell = cells[cellIdx];
+        startCell.distance = 0;
+
+        let maxDist = 0;
+        let frontierCells = [startCell];
+
+        while (frontierCells.length) {
```

Comment:
yay you did BFS with a while loop!!!! Weren't we talking about how you usually do it with recursion haha?